### ISSUE TITLE
Feature request: custom panic handling 

### ISSUE THREAD

**Issue by xmclark** [bastion-rs/bastion#33]
Attaching a custom panic handler would be useful in applications that want to run logging or cleanup code on catastrophic failures. 

The panic handler shouldn’t interfere with the rusts built in panic handling solution.

**Comment by vertexclique** [comment]
This is doable, for every failure will trigger a global handler closure.

**Comment by xmclark** [comment]
Yeah I don't want to suggest the exact implementation. I'm more interested in the behavior of understanding when panics happen, and being able to act with some basic context of the failure. 

My use case:
Running some user code (perhaps some event processors) that may panic. On failures, I need to ensure that I perform some cleanup up make a few service requests to tell other systems about failures/nack. 

I tried making this work before with Bastion, but did not find a nice solution, so I just rolled-my-own supervision.
